src/harness/externalCompileRunner.ts

```diff
@@ -33,18 +37,34 @@ abstract class ExternalCompileRunnerBase extends RunnerBase {
             const cp = require("child_process");
 
             it("should build successfully", () => {
-                const cwd = path.join(__dirname, "../../", this.testDir, directoryName);
+                let cwd = path.join(__dirname, "../../", this.testDir, directoryName);
                 const timeout = 600000; // 600s = 10 minutes
+                const stdio = isWorker ? "pipe" : "inherit";
+                let types: string[];
+                if (fs.existsSync(path.join(cwd, "test.json"))) {
+                    const update = cp.spawnSync("git", ["submodule", "update", "--remote"], { cwd, timeout, shell: true, stdio });
+                    if (update.status !== 0) throw new Error(`git submodule update for ${directoryName} failed!`);
+
+                    const config = JSON.parse(fs.readFileSync(path.join(cwd, "test.json"), { encoding: "utf8" })) as UserConfig;
+                    ts.Debug.assert(!!config.types, "Git is the only reason for using test.json right now");

```

Comment:
`"Git is the only reason for using test.json right now"`? That's not what's being checked here, so it's odd. I think `"Types field must be present"` would be sufficient. And likely not need to change in the future.